File: App/app/src/main/java/com/github/travelplannerapp/login/LoginModule.java
package com.github.travelplannerapp.login;

import dagger.Module;
import dagger.Provides;

/**
 * Define LoginActivity-specific dependencies here.
 */
@Module
public class LoginModule {

    @Provides
    LoginPresenter provideLoginPresenter(LoginContract.View loginView) {
        return new LoginPresenter(loginView);
    }
}

File: App/app/src/main/java/com/github/travelplannerapp/login/LoginModule.kt
package com.github.travelplannerapp.login

import dagger.Module
import dagger.Provides

/**
 * Define LoginActivity-specific dependencies here.
 */
@Module
class LoginModule {

    @Provides
    internal fun provideLoginPresenter(loginView: LoginContract.View): LoginPresenter {
        return LoginPresenter(loginView)
    }
}